docs/CInterop.md

```diff
@@ -342,3 +343,44 @@ out/main.c:9153:29: error: invalid operands to binary expression ('String' (aka
 1 error generated.
 ```
 
+## Callbacks
+
+Some C APIs rely on callbacks, let's define a C function that accepts a
+callback and an argument and returns the result of calling that function as an
+example:
+
+```clojure
+(deftemplate runner (Fn [(Ptr ()) (Ptr ())] a)
+                    "$a $NAME(void* fn, void* args)"
+                    "$a $NAME(void* fnptr, void* args) {
```

Comment:
Maybe make the argument names for the function pointer the same in the signature string and declaration?

Reply:
Not sure why I didn't use `$DECL` here really.